src/ui/map.js

```diff
@@ -2426,6 +2486,17 @@ class Map extends Camera {
         this._canvas.style.height = `${height}px`;
     }
 
+    _addMarker(marker: Marker) {
+        this._markers.push(marker);
```

Comment:
It seems a bit messy to have some marker changes happen by the marker listening to map events and other changes to happen by the map calling methods directly on them.

Reply:
But also don't have a clear suggestion here.